src/spotify_to_tidal/search.py

```diff
@@ -0,0 +1,30 @@
+import tidalapi
+import tidalapi.session
+
+from cachetools import TTLCache, cached
+
+def simple(input_string: str) -> str:
+    # only take the first part of a string before any hyphens or brackets to account for different versions
+    return input_string.split('-')[0].strip().split('(')[0].strip().split('[')[0].strip()
+
+# We simplify the search to easily cache results
+
+@cached(TTLCache(maxsize=1024, ttl=600))
+def _search_tidal(
+    tidal_session: tidalapi.Session,
+    q: str,
+    model: tidalapi.Album | tidalapi.Track | tidalapi.Artist
+    ) -> tidalapi.session.SearchResults:
+    return tidal_session.search(q, models=[model])
+
+@cached(TTLCache(maxsize=1024, ttl=600))
+def search_tidal_albums(tidal_session: tidalapi.Session, artist: str, album: str):
+    artist = simple(artist.casefold())
+    album = simple(album.casefold())
+    return _search_tidal(tidal_session, artist+" "+album, tidalapi.Album)
+
+@cached(TTLCache(maxsize=1024, ttl=600))
```

Comment:
Did you do some tuning on these numbers for size and time? I would think that these numbers might be too small for large collections...?

Reply:
In fact I'm not convinced that using cachetools is all that helpful here. When I tried running it I didn't observe many cache hits, as each subprocess has a separate cache.

Reply:
Sounds good. The numbers I chose were random.